ambry-clustermap/src/main/java/com/github/ambry/clustermap/DataNodeConfig.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+
+package com.github.ambry.clustermap;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * A data object for configs scoped to a single data node.
+ */
+class DataNodeConfig {
+  private final String instanceName;
+  private final String hostName;
+  private final int port;
+  private final String datacenterName;
+  private final Integer sslPort;
+  private final Integer http2Port;
+  private final String rackId;
+  private final long xid;
+  private final Set<String> sealedReplicas = new HashSet<>();
+  private final Set<String> stoppedReplicas = new HashSet<>();
```

Comment:
I will introduce another list of `disabledReplicas` in InstanceConfig (and will be migrated to DataNodeConfig in near future). Could you add a set of disabled replicas as a placeholder here?  For now it should be empty and no router logic depends on it. The purpose of `disabledReplicas` is to allow server to disable replicas on bad disk (also triggers state transition) and to automatically re-enable these replicas when disk is replaced and server is restarted.

Reply:
Yes, that's true, it will be cleared if store starts successfully (which means disk is good or replaced). If you need more context, we can discuss it offline.

Reply:
Added a placeholder for disabled replicas